lib/SampleService/SampleServiceImpl.py

```diff
@@ -197,6 +197,125 @@ def create_sample(self, ctx, params):
+           { "name": "Castle Geyser", "lat": 44.463816, "long": -110.836471 }
+           "primitive values" means an int, float, string, or equivalent
+           typedefs. Including any collection types is an error.) -> mapping
+           from type "metadata_value_key" (A key for a value associated with
+           a piece of metadata. Less than 1000 unicode characters. Examples:
+           units, value, species) to unspecified object, parameter
+           "meta_user" of type "metadata" (Metadata attached to a sample.) ->
+           mapping from type "metadata_key" (A key in a metadata key/value
+           pair. Less than 1000 unicode characters.) to type "metadata_value"
+           (A metadata value, represented by a mapping of value keys to
+           primitive values. An example for a location metadata key might be:
+           { "name": "Castle Geyser", "lat": 44.463816, "long": -110.836471 }
+           "primitive values" means an int, float, string, or equivalent
+           typedefs. Including any collection types is an error.) -> mapping
+           from type "metadata_value_key" (A key for a value associated with
+           a piece of metadata. Less than 1000 unicode characters. Examples:
+           units, value, species) to unspecified object, parameter "name" of
+           type "sample_name" (A sample name. Must be less than 255
+           characters.), parameter "save_date" of type "timestamp" (A
+           timestamp in epoch milliseconds.), parameter "version" of type
+           "version" (The version of a sample. Always > 0.), parameter
+           "prior_version" of Long, parameter "as_admin" of type "boolean" (A
+           boolean value, 0 for false, 1 for true.), parameter "as_user" of
+           type "user" (A user's username.)
+        :returns: instance of list of type "SampleAddress" (A Sample ID and
+           version. id - the ID of the sample. version - the version of the
+           sample.) -> structure: parameter "id" of type "sample_id" (A
+           Sample ID. Must be globally unique. Always assigned by the Sample
+           service.), parameter "version" of type "version" (The version of a
+           sample. Always > 0.)
+        """
+        # ctx is the context object
+        # return variables are: addresses
+        #BEGIN create_samples
+        if not params.get('samples'):
+          raise ValueError(f"")
+        as_admin, user = _get_admin_request_from_object(params, 'as_admin', 'as_user')
+        addresses = []
+        _check_admin(
+            self._user_lookup, ctx[_CTX_TOKEN], _AdminPermission.FULL,
+            # pretty annoying to test ctx.log_info is working, do it manually
+            'create_samples', ctx.log_info, as_user=user, skip_check=not as_admin)
+        for samp in params['samples']:
+          sample, id_, prev_ver = _create_sample_params(samp)
+          ret = self._samples.save_sample(
```

Comment:
I think we could do a `try catch` here. Then add failed samples to something like `failed_addresses`?